packages/vite-plugin-cloudflare/src/cloudflare-environment.ts

```diff
@@ -124,11 +124,21 @@ export const cloudflareBuiltInModules = [
 const defaultConditions = ["workerd", "worker", "module", "browser"];
 const target = "es2022";
 
-export function createCloudflareEnvironmentOptions(
-	workerConfig: WorkerConfig,
-	userConfig: vite.UserConfig,
-	environment: { name: string; isEntry: boolean }
-): vite.EnvironmentOptions {
+export function createCloudflareEnvironmentOptions({
+	workerConfig,
+	userConfig,
+	mode,
+	environmentName,
+	isEntryWorker,
+}: {
+	workerConfig: WorkerConfig;
+	userConfig: vite.UserConfig;
+	mode: string;
```

Comment:
I'm not sure if it makes things much better, but potentially this could be: `mode: vite.ConfigEnv['mode'];` to indicate more clearly that we're reusing `mode` from vite? 🤔

Reply:
The same applies to `getProcessEnvReplacements`

Reply:
https://github.com/cloudflare/workers-sdk/pull/10422/commits/d6422944217afd95f8976fc690aa4620e162cc6c